aiogram/dispatcher/event/handler.py

```diff
@@ -37,6 +38,8 @@ def _prepare_kwargs(self, kwargs: Dict[str, Any]) -> Dict[str, Any]:
         }
 
     async def call(self, *args: Any, **kwargs: Any) -> Any:
+        if "FSMContextMiddleware.close" in str(self.callback):
```

Comment:
wtf is going on?

Reply:
I didn't better solution for this bug.

Reply:
It is very bad solution and doesn't make anything better.
You should not change this place, your issue can be fixed in two lines at place where which you do `.pop`.

Just check that key is present in workflow_data and only then pop it, but all other changes should be rolled back.